pkg/test/cel/backendtlspolicy_test.go

```diff
@@ -0,0 +1,198 @@
+	"testing"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	gatewayv1a2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+	"sigs.k8s.io/gateway-api/apis/v1beta1"
+)
+
+func TestBackendTLSPolicyConfig(t *testing.T) {
+	tests := []struct {
+		name        string
+		wantErrors  []string
+		routeConfig gatewayv1a2.BackendTLSPolicyConfig
+	}{
+		{
+			name:        "invalid BackendTLSPolicyConfig with missing fields",
+			routeConfig: gatewayv1a2.BackendTLSPolicyConfig{},
+			wantErrors:  []string{"spec.tls.hostname in body should be at least 1 chars long", "must specify either CACertRefs or WellKnownCACerts"},
+		},
+		{
+			name: "valid BackendTLSPolicyConfig",
+			routeConfig: gatewayv1a2.BackendTLSPolicyConfig{
+				WellKnownCACerts: ptrTo(gatewayv1a2.WellKnownCACertType("System")),
+				Hostname:         "foo.example.com",
+			},
+			wantErrors: []string{},
+		},
+		{
+			name: "invalid BackendTLSPolicyConfig with both CACertRefs and WellKnownCACerts",
+			routeConfig: gatewayv1a2.BackendTLSPolicyConfig{
+				CACertRefs: []v1beta1.LocalObjectReference{
+					{
+						Group: "group",
+						Kind:  "kind",
+						Name:  "name",
+					},
+				},
+				WellKnownCACerts: ptrTo(gatewayv1a2.WellKnownCACertType("System")),
+				Hostname:         "foo.example.com",
+			},
+			wantErrors: []string{"must not contain both CACertRefs and WellKnownCACerts"},
+		},
+		{
+			name: "invalid BackendTLSPolicyConfig with missing hostname field",
+			routeConfig: gatewayv1a2.BackendTLSPolicyConfig{
+				CACertRefs: []v1beta1.LocalObjectReference{
+					{
+						Group: "group",
+						Kind:  "kind",
+						Name:  "name",
+					},
+				},
+			},
+			wantErrors: []string{"spec.tls.hostname in body should be at least 1 chars long"},
+		},
+		{
+			name: "invalid BackendTLSPolicyConfig with missing caCertRefs[0].name and caCertRefs[0].kind field",
+			routeConfig: gatewayv1a2.BackendTLSPolicyConfig{
+				CACertRefs: []v1beta1.LocalObjectReference{
+					{
+						Group: "group",
+					},
+				},
+				Hostname: "foo.example.com",
+			},
+			wantErrors: []string{"spec.tls.caCertRefs[0].name in body should be at least 1 chars long", "spec.tls.caCertRefs[0].kind in body should be at least 1 chars long"},
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			route := &gatewayv1a2.BackendTLSPolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      fmt.Sprintf("foo-%v", time.Now().UnixNano()),
+					Namespace: metav1.NamespaceDefault,
+				},
+				Spec: gatewayv1a2.BackendTLSPolicySpec{
+					TargetRef: gatewayv1a2.PolicyTargetReferenceWithSectionName{
+						PolicyTargetReference: gatewayv1a2.PolicyTargetReference{
+							Group:     "group",
+							Kind:      "kind",
+							Name:      "name",
+							Namespace: ptrTo(gatewayv1a2.Namespace("ns")),
+						},
+					},
+					TLS: tc.routeConfig,
+				},
+			}
+			validateBackendTLSPolicy(t, route, tc.wantErrors)
+		})
+	}
+}
+
+func TestPolicyTargetReferenceWithSectionName(t *testing.T) {
```

Comment:
nit: 

More tests are definitely good to have, but maybe we should not mix non-CEL validations (`spec.targetRef.name in body should be at least 1 chars long` and others) and CEL validations errors in these tests?

I can be corrected, but I believe we have https://github.com/kubernetes-sigs/gateway-api/tree/main/hack/invalid-examples for non-CEL validation testing.